npm/ux/scatterChart.tsx

```diff
@@ -201,199 +85,216 @@ export function ScatterChart(props: {
   const plotAreaHeight = svgHeight - yMargin;
 
-  const viewBox = `${svgXMin - svgViewBoxWidthPadding} ${
-    -plotAreaHeight - svgViewBoxHeightPadding
-  } ${svgWidth + svgViewBoxWidthPadding} ${
-    svgHeight + svgViewBoxHeightPadding
-  }`;
+  const viewBox = `${svgXMin} ${-plotAreaHeight - svgViewBoxHeightPadding} ${
+    svgWidth + svgViewBoxWidthPadding
+  } ${svgHeight + svgViewBoxHeightPadding}`;
 
   const yAxisTextPaddingFromTicks = 5;
   const yAxisTextYPadding = 4;
 
-  function trySetAttribute(element: Element, attribute: string, value: string) {
-    if (element.getAttribute(attribute) !== value) {
-      element.setAttribute(attribute, value);
-    }
+  function renderTooltip(
+    topDiv: HTMLDivElement,
+    point: SVGCircleElement,
+    tooltip: HTMLDivElement,
+  ) {
+    const label = point.getAttribute("data-label");
+    tooltip.textContent = label;
+    const halfWidth = tooltip.offsetWidth / 2;
+    const pointRect = point.getBoundingClientRect();
+    const centerY = (pointRect.top + pointRect.bottom) / 2;
+    const centerX = (pointRect.left + pointRect.right) / 2;
+    const divRect = topDiv.getBoundingClientRect();
+    tooltip.style.left = `${centerX - divRect.left - halfWidth}px`;
+    tooltip.style.top = `${centerY - divRect.top + 12}px`;
+    tooltip.style.visibility = "visible";
   }
 
-  function hideHoverTooltip() {
-    const tooltip = chart?.querySelector("#tooltip-hover");
-
-    if (tooltip) {
-      if (tooltip.getAttribute("clicked") === "false") {
-        tooltip.setAttribute("visibility", "hidden");
-      }
+  function onPointMouseEvent(ev: MouseEvent, eventType: string) {
+    // Ensure we have a point as the target
+    if (!(ev.target instanceof SVGCircleElement)) return;
+    const target = ev.target as SVGCircleElement;
+    if (!target.classList.contains("qs-scatterChart-point")) return;
+
+    // Get the div enclosing the chart, and the popup child of it.
+    const topDiv = target.closest("div") as HTMLDivElement;
+    const popup = topDiv.querySelector(
+      ".qs-scatterChart-tooltip",
+    ) as HTMLDivElement;
+
+    switch (eventType) {
+      case "over":
+        {
+          renderTooltip(topDiv, target, popup);
+        }
+        break;
+      case "out":
+        popup.style.visibility = "hidden";
+        break;
+      case "click":
+        {
+          if (target.classList.contains("qs-scatterChart-point-selected")) {
+            // Clicked on the already selected point, so delete the point/row
+            props.onPointSelected(-1, 0);
+          } else {
+            const index = JSON.parse(target.getAttribute("data-index")!);
+            props.onPointSelected(index[0], index[1]);
+          }
+        }
+        break;
+      default:
+        console.error("Unknown event type: ", eventType);
     }
   }
 
-  const scatterChartContainerRef = useRef<SVGSVGElement | null>(null);
-  let chart: Element | null = null;
-
-  function updateCoordinates() {
```

Comment:
This one was recently introduced to consolidate all coordinate computations and separate them from rendering (which was originally provided).

A potential for this was a chart resizing scenario. Let us discuss if we would like to consider the resizing scenario and if so, if the proposed update would support it.

Reply:
Resizing should update some state (either stored in the passed in props or locally with useState), which would cause a re-render, so the new approach would work fine with resizing.